### PR TITLE
Fixed#31558: Added support for the use of boolean attribute on properties in the admin.

### PR DESCRIPTION
**Pull request by RyanWalker277**
A possible fix for [#31558](https://code.djangoproject.com/ticket/31558). Not entirely sure if it's the absolute right way to proceed with the issue. Wanted some feedback on my approach to improve things further.

### CODE REVIEW

File: django/contrib/admin/templatetags/admin_list.py [hunk 1]
```diff
@@ -224,8 +239,7 @@ def link_in_col(is_first, field_name, cl):
             if f is None or f.auto_created:
                 if field_name == "action_checkbox":
                     row_classes = ["action-checkbox"]
-                boolean = getattr(attr, "boolean", False)
-                result_repr = display_for_value(value, empty_value_display, boolean)
+                result_repr = get_result_repr(attr, value, empty_value_display)
```
Comment (felixxm) [inline django/contrib/admin/templatetags/admin_list.py:242]: I'm not sure why you want to add two nested helpers here :thinking: Do we need them? It should be enough to check `boolean` for a property, e.g.
```diff
diff --git a/django/contrib/admin/templatetags/admin_list.py b/django/contrib/admin/templatetags/admin_list.py
index 9bff84d501..498e27e77d 100644
--- a/django/contrib/admin/templatetags/admin_list.py
+++ b/django/contrib/admin/templatetags/admin_list.py
@@ -225,6 +225,9 @@ def items_for_result(cl, result, form):
                 if field_name == "action_checkbox":
                     row_classes = ["action-checkbox"]
                 boolean = getattr(attr, "boolean", False)
+                # Set boolean for attr that is a property, if defined.
+                if isinstance(attr, property) and hasattr(attr, 'fget'):
+                    boolean = getattr(attr.fget, 'boolean', False)
                 result_repr = display_for_value(value, empty_value_display, boolean)
                 if isinstance(value, (datetime.date, datetime.time)):
                     row_classes.append("nowrap")
```

File: tests/modeladmin/test_actions.py [hunk 2]
```diff
@@ -148,3 +149,23 @@ def custom_action_2(self, request, queryset):
...
+        for result in results:
+            if "action-checkbox" in result:
+                found_action_checkbox = True
+
+            if "field-__str__" in result:
+                self.assertIn("Test Model", result)
+                found_name_field = True
+
+        self.assertTrue(found_name_field)
+        self.assertTrue(found_action_checkbox)
```
Comment (felixxm) [inline tests/modeladmin/test_actions.py:171]: Can we use `all()` instead?

**Review by felixxm — COMMENTED** [review]
@RyanWalker277 Thanks for this patch :+1: Can we also add a short example in docs?
https://github.com/django/django/blob/68912e4f6f84f21322f92a2c7b6c77f68f91b9c9/docs/ref/contrib/admin/index.txt#L812-L821

File: django/contrib/admin/templatetags/admin_list.py (hunk 1 above)
Comment (RyanWalker277) [reply django/contrib/admin/templatetags/admin_list.py]: Yeah right! The two nested helpers were making it overly complex. Thanks for pointing this out.

File: tests/modeladmin/test_actions.py (hunk 2 above)
Comment (RyanWalker277) [reply tests/modeladmin/test_actions.py]: Yeah, I have modified the test to use `all()` instead of the existing loop assertions.

**Comment by RyanWalker277** [comment]
> @RyanWalker277 Thanks for this patch +1 Can we also add a short example in docs?
> 
> https://github.com/django/django/blob/68912e4f6f84f21322f92a2c7b6c77f68f91b9c9/docs/ref/contrib/admin/index.txt#L812-L821

Sure! I am a bit unsure how to update the existing docs for this change. For now, I have added a line to the existing sample. Please have a look and let me know if it needs to be written a bit differently. Any feedback is much appreciated :open_hands:

File: tests/modeladmin/test_actions.py [hunk 3]
```diff
@@ -148,3 +149,20 @@ def custom_action_2(self, request, queryset):
                 "Local admin action 2.",
             ],
         )
+
+    def test_items_for_result(self):
```
[the remark this answers is not in the document]
Comment (felixxm) [reply tests/modeladmin/test_actions.py:153]: Also, it seems that it doesn't work as icons are not in the response. Have you tested it on a simple project?

**Comment by felixxm** [comment]
@RyanWalker277 Do you have time to keep working on this?

**Comment by RyanWalker277** [comment]
Hey @felixxm, I am working on the patch. Recently I was engaged somewhere else so didn't have the time to pick it up. Will be doing the changes soon.

**Comment by felixxm** [comment]
Closing due to inactivity.

File: tests/modeladmin/test_actions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (RyanWalker277) [reply tests/modeladmin/test_actions.py:153]: I haven't tested on a project. Also, I am having a hard time getting the icons to show in the response. I can use some help here to debug where I am going wrong.

**Pull request closed without merging**